consensus/src/experimental/buffer_item.rs

```diff
@@ -273,29 +350,35 @@ impl BufferItem {
     pub fn try_advance_to_aggregated(self, validator: &ValidatorVerifier) -> Self {
         match self {
             Self::Signed(signed_item) => {
-                if signed_item
-                    .commit_proof
-                    .check_voting_power(validator)
+                if validator
+                    .check_voting_power(signed_item.partial_commit_proof.signatures().keys())
                     .is_ok()
                 {
                     Self::Aggregated(Box::new(AggregatedItem {
                         executed_blocks: signed_item.executed_blocks,
-                        commit_proof: signed_item.commit_proof,
+                        commit_proof: aggregate_commit_proof(
+                            signed_item.partial_commit_proof.ledger_info(),
+                            signed_item.partial_commit_proof.partial_sigs(),
+                            validator,
+                        ),
                         callback: signed_item.callback,
                     }))
                 } else {
                     Self::Signed(signed_item)
                 }
             }
             Self::Executed(executed_item) => {
-                if executed_item
-                    .commit_proof
-                    .check_voting_power(validator)
+                if validator
+                    .check_voting_power(executed_item.partial_commit_proof.signatures().keys())
                     .is_ok()
                 {
                     Self::Aggregated(Box::new(AggregatedItem {
                         executed_blocks: executed_item.executed_blocks,
-                        commit_proof: executed_item.commit_proof,
+                        commit_proof: aggregate_commit_proof(
```

Comment:
this is not safe since it panics on bad signatures but we carry unverified signatures from ordered to executed